src/refresh.rs

```diff
@@ -0,0 +1,79 @@
+use std::{
+    os::windows::prelude::OsStrExt,
+    ffi::OsStr,
+    thread::sleep,
+    time::Duration,
+    ptr::null_mut,
+    iter::once
+};
+
+use windows::Win32::{
+    Foundation::{
+        HWND,
+        LPARAM,
+        WPARAM,
+        BOOL
+    },
+    UI::WindowsAndMessaging::{
+        WM_THEMECHANGED,
+        EnumWindows,
+        WM_SETTINGCHANGE,
+        SendMessageTimeoutW,
+        SMTO_BLOCK,
+        SMTO_NOTIMEOUTIFNOTHUNG
+    }
+};
+
+use crate::regkey::{
+    RegistryKey,
+    RegistryPermission
+};
+
+fn os_str(value: &str) -> Vec<u16> {
+    OsStr::new(value).encode_wide().chain(once(0)).collect()
+}
+
+unsafe extern "system" fn refresh_window_callback(hwnd: HWND, _: LPARAM) -> BOOL {
+    SendMessageTimeoutW(
+        hwnd,
+        WM_SETTINGCHANGE,
+        WPARAM(0),
+        LPARAM(os_str("ImmersiveColorSet").as_ptr() as isize),
+        SMTO_BLOCK | SMTO_NOTIMEOUTIFNOTHUNG,
+        200,
+        null_mut()
+    );
+
+    SendMessageTimeoutW(
+        hwnd,
+        WM_THEMECHANGED,
+        WPARAM(0),
+        LPARAM(0),
+        SMTO_BLOCK | SMTO_NOTIMEOUTIFNOTHUNG,
+        200,
+        null_mut()
+    );
+    
+    BOOL(1)
+}
+
+pub fn refresh_windows() {
+    sleep(Duration::from_millis(20));
```

Comment:
what is that for? this is a weird thing to have as the first line in a function haha

Reply:
This line is mostly preventive. It was left out from previous attempts. It can be removed.